Cookbook/Technical-Documents/From_0%_to_100%_with_Bento.md

```diff
@@ -0,0 +1,443 @@
+
+### Examples from our code
+Provide examples of our app & code what screens are written in Bento and link them here as a reference point 
+
+### Bento & SwiftUI
+Small explanation on what our team's current approach to adopting SwiftUI looks like. This will be a be a short section saying that Bento will eventually replaced by SwiftUI (in few years perspective).
+
+## How to use BabylonBoxViewController, ViewModel & Renderer
+Section when I'd like to describe how to setup a typical RAF module with strong focus on `Renderer` part.
+I want this section to be a guide with code samples, line by line, how to write a renderer.
+
+### Keywords in Bento
+- What are `Screen`, `Box`, `Section`, `Node`.
+- What is a `BabylonBoxViewController`?
+- What is `AnyRenderable`
+- What Operators can I use?
+
+## How to write a component which is not in Design Library 
+**Introduction** to Design Library. Why is it there and how is it related with Bento? How do I find a component I need to use? What to do if I cannot find a component?
+
+## BabylonBoxViewController from scratch (optional)
+Note: I'm still thinking if that should be a part of the article.
+Rarely, but sometimes, we cannot use `BabylonBoxViewController`. I'd like to guide how a simple implementation of custom ViewController can look like. Maybe, the example would use UICollectionView instead of UITableView which is used by `BabylonBoxViewController`
+
+## Outline notes
+- The order of section may change in the article. I'd like to make this article about 1k words long. 
+- Naming of sections may be changed in the future (to sound better)
+- Sections marked as `(optional)` may be skipped if I find the article gets too big and overwhelming.
+- Once it's accepted by @scass91 and @RuiAAPeres I'll change the PR to be "draft" and I'm going to work on the article in this file below the `----`. Before the final merge I'm going to delete the outline.
+
+---- 
+# The Article (in progress):
+To be added... Stay tuned 😎
+
+In this article I would answer few questions regarding Bento. If you are a new person in the team and you have questions like: "What is Bento?", "Why do we use it?", "How should I use it?", "What's a renderable?". If you've had any of these questions, then the article is for you.
+
+## What and why?
+First of all, Bento is our internal library which allows us to write UI code in a declarative way. This makes it faster to write the UI code. 
+
+```plain
+Bento now is our internal library however we used to open source it. It has changed when Apple announced SwiftUI at WWDC 2019. We decided to move it under the main repo back then.
+```
+
+## This is what we want to build
+![](/Users/adam.borek/Developer/babylon/images/bento01.png)
+
+## Renderer
+I assume you are already kind of familiar with our Architecture. Quick recap, a typical screen is built from:
+- ViewController
+- ViewModel
+- Renderer
+- Builder
+- FlowController
+
```

Comment:
Would it be worth having these assumptions at the top of the article so a reader is familiar with what they should know from the start?